notebooks/115-async-api/115-async-api.ipynb

```diff
@@ -0,0 +1,460 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def preprocess(image):\n",
+    "    \"\"\"\n",
+    "    Define the preprocess function for input data\n",
+    "    \n",
+    "    :param: image: the orignal input frame\n",
+    "    :returns:\n",
+    "            resized_image: the image processed\n",
+    "    \"\"\"\n",
+    "    resized_image = cv2.resize(image, shape)\n",
+    "    resized_image = cv2.cvtColor(np.array(resized_image), cv2.COLOR_BGR2RGB)\n",
+    "    resized_image = resized_image.transpose((2, 0, 1))\n",
+    "    resized_image = np.expand_dims(resized_image, axis=0).astype(np.float32)\n",
+    "    return resized_image\n",
+    "\n",
+    "\n",
+    "def postprocess(result, image, fps):\n",
+    "    \"\"\"\n",
+    "    Define the postprocess function for output data\n",
+    "    \n",
+    "    :param: result: the inference results\n",
+    "            image: the orignal input frame\n",
+    "            fps: average throughput calculated for each frame\n",
+    "    :returns:\n",
+    "            image: the image with bounding box and fps message\n",
+    "    \"\"\"\n",
+    "    detections = result.reshape(-1, 7)\n",
+    "    for i, detection in enumerate(detections):\n",
+    "        _, image_id, confidence, xmin, ymin, xmax, ymax = detection\n",
+    "        if confidence > 0.5:\n",
+    "            xmin = int(max((xmin * image.shape[1]), 10))\n",
+    "            ymin = int(max((ymin * image.shape[0]), 10))\n",
+    "            xmax = int(min((xmax * image.shape[1]), image.shape[1] - 10))\n",
+    "            ymax = int(min((ymax * image.shape[0]), image.shape[0] - 10))\n",
+    "            cv2.rectangle(image, (xmin, ymin), (xmax, ymax), (0, 255, 0), 2)\n",
+    "            cv2.putText(image, str(round(fps, 2)) + \" fps\", (5, 20), cv2.FONT_HERSHEY_SIMPLEX, 0.7, (0, 255, 0), 3) \n",
+    "    return image"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Get the test video"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "video_path = \"../202-vision-superresolution/data/CEO Pat Gelsinger on Leading Intel.mp4\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## How to improve the throughput of video processing\n",
+    "We can compare the performace for the regular and async-based approaches:\n",
+    "\n",
+    "### Sync Mode (regular approach)\n",
+    "Let's see how video processing works in the regular way and it's pseudo-code.<br />\n",
+    "In the regular way, the frame is captured with OpenCV and then immediately processed:\n",
+    "\n",
+    "<img align='center' src=\"https://user-images.githubusercontent.com/91237924/168452573-d354ea5b-7966-44e5-813d-f9053be4338a.png\" alt=\"drawing\" width=\"1000\"/><br />\n",
```

Comment:
Is it possible to improve the quality of this image? It seems a little bit blurry